controllers/services_controller.go

```diff
@@ -0,0 +1,181 @@
+	"kubevirt.io/ssp-operator/internal/operands/metrics"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	MetricsServiceName = "ssp-operator-metrics"
+	OperatorName       = "ssp-operator"
+)
+
+func ServiceObject(namespace string) *v1.Service {
+	policyCluster := v1.ServiceInternalTrafficPolicyCluster
+	familyPolicy := v1.IPFamilyPolicySingleStack
+	return &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      MetricsServiceName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				metrics.PrometheusLabelKey: metrics.PrometheusLabelValue,
+			},
+		},
+		Spec: v1.ServiceSpec{
+			InternalTrafficPolicy: &policyCluster,
+			IPFamilies:            []v1.IPFamily{v1.IPv4Protocol},
+			IPFamilyPolicy:        &familyPolicy,
+			Ports: []v1.ServicePort{
+				{
+					Name:       metrics.MetricsPortName,
+					Port:       443,
+					Protocol:   v1.ProtocolTCP,
+					TargetPort: intstr.FromString(metrics.MetricsPortName),
+				},
+			},
+			Selector: map[string]string{
+				metrics.PrometheusLabelKey: metrics.PrometheusLabelValue,
+				"name":                     OperatorName,
+			},
+			SessionAffinity: v1.ServiceAffinityNone,
+			Type:            v1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+// Annotation to generate RBAC roles to read and modify services
+// +kubebuilder:rbac:groups="",resources=services,verbs=get;watch;list;create;update;delete
+
+func CreateServiceController(mgr ctrl.Manager) (*serviceReconciler, error) {
+	return newServiceReconciler(mgr)
+}
+
+func (r *serviceReconciler) Start(ctx context.Context, mgr ctrl.Manager) error {
+	err := r.createMetricsService(ctx, mgr.GetLogger())
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("error start serviceReconciler: %w", err)
+	}
+
+	return r.setupController(mgr)
+}
+
+func (r *serviceReconciler) setServiceOwnerReference(service *v1.Service) error {
+	return controllerutil.SetOwnerReference(r.deployment, service, r.client.Scheme())
+}
+
+func (r *serviceReconciler) createMetricsService(ctx context.Context, logger logr.Logger) error {
+	service := ServiceObject(r.serviceNamespace)
+	err := r.setServiceOwnerReference(service)
+	if err != nil {
+		return fmt.Errorf("error setting owner reference: %w", err)
+	}
+	return r.client.Create(ctx, service)
+}
+
+func (r *serviceReconciler) setupController(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("service-controller").
+		For(&v1.Service{}).
+		Complete(r)
+}
+
+// serviceReconciler reconciles the required services in the operator's namespace
+type serviceReconciler struct {
+	client           client.Client
+	apiReader        client.Reader
+	log              logr.Logger
+	serviceNamespace string
+	deployment       *apps.Deployment
+}
+
+func (r *serviceReconciler) setDeploymentField() error {
```

Comment:
Can you make this a free function that takes `apiReader` parameter and returns `(*Deployment, error)` ?
Then the `apiReader` does not need to be in `serviceReconciler` struct.